cpp/src/integrations/cudf/partition.cpp

```diff
@@ -238,25 +260,24 @@ std::vector<PackedData> unspill_partitions(
         }
     }
 
-    // This total sum is what we need to reserve before moving data to device.
-    auto reservation =
-        br->reserve_and_spill(MemoryType::DEVICE, non_device_size, allow_overbooking);
+    return with_memory_reservation(
+        br->reserve_and_spill(MemoryType::DEVICE, non_device_size, allow_overbooking),
+        [&](auto& reservation) {
+            // Unspill each partition.
+            std::vector<PackedData> ret;
+            ret.reserve(partitions.size());
+            for (auto& [metadata, data] : partitions) {
+                ret.emplace_back(
+                    std::move(metadata), br->move(std::move(data), stream, reservation)
+                );
+            }
 
-    return with_memory_reservation(std::move(reservation), [&] {
-        // Unspill each partition.
-        std::vector<PackedData> ret;
-        ret.reserve(partitions.size());
-        for (auto& [metadata, data] : partitions) {
-            ret.emplace_back(
-                std::move(metadata), br->move(std::move(data), stream, reservation)
+            statistics->add_duration_stat(
+                "spill-time-host-to-device", Clock::now() - elapsed
```

Comment:
`elapsed` is a bad name, it should be `start_time` or something similar. Elapsed would be `Clock::now() - start_time`, but this is just using how it was already named before this PR, so this is fine.